### PR TITLE
[SageMaker] Allow passing of SageMaker Estimator arguments during job…

### LINKED ISSUE #309: [Bug] SageMaker-based training and inference does not support --volume-size argument
The SageMaker training and inference instances have 30GB disk volume by default. 

But, for large graphs, e.g., MAG-large, one partition will be much larger than 30GB. And will cause "lack of disk space" error when download file from S3.

Therefore, need to set the SageMaker --volume-size argument and pass it in the Training and Inference launch scripts.

### PR DESCRIPTION
**Pull request by thvasilo**
… launch

*Issue #, if available:*
Fixes #309 

*Description of changes:*
Allow forwarding of SageMaker Estimator arguments during job launch.

To achieve this we create a common parser for all launch scripts, that includes a new argument `sm-estimator-parameters`. End users can provide a space-separated list of SageMaker Estimator arguments here, in the format
`arg_name1=arg_val1 arg_name2=arg_val2`.

We use space as the entry separator, and the values themselves must not contain extra spaces. 

So a list of strings needs to be passed as `subnets=['subnet-123456','subnet-567799']` and not `subnets=['subnet-123456', 'subnet-567799']`.
We note this detail in the README. 

We parse the values as Python literals allowing them to interpreted with the expected type (str, int, list(str), bool) 

Tested by running SageMaker jobs where we pass e.g. `--sm-estimator-parameters "max_run=3600 volume_size=100 encrypt_inter_container_traffic=True subnets=['subnet-123456','subnet-567799']"`

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: sagemaker/launch/launch_gconstruct.py [hunk 1]
```diff
@@ -95,44 +95,38 @@ def run_job(input_args, image, unknowargs):
...
 
 if __name__ == "__main__":
-    arg_parser = parse_args()
+    arg_parser = get_gconstruct_parser()
     args, unknownargs = arg_parser.parse_known_args()
     print(args)
 
     gconstruct_image = args.image_url
 
+    if not args.instance_type:
```
Comment (classicsong) [inline sagemaker/launch/launch_gconstruct.py:129]: In common_parser L23 we force users to provide an instance_type. So this if statement will always be False.

File: sagemaker/launch/launch_infer.py [hunk 2]
```diff
@@ -71,7 +71,7 @@ def run_job(input_args, image, unknowargs):
 
     container_image_uri = image
 
-    prefix = "script-mode-container"
+    prefix = f"graphstorm-train-{sm_task_name}"
```
Comment (classicsong) [inline sagemaker/launch/launch_infer.py:74]:
```suggestion
    prefix = f"graphstorm-infer-{sm_task_name}"
```

File: sagemaker/launch/launch_infer.py [hunk 3]
```diff
@@ -125,61 +125,58 @@ def run_job(input_args, image, unknowargs):
...
     return parser
 
 if __name__ == "__main__":
-    arg_parser = parse_args()
+    arg_parser = get_inference_parser()
     args, unknownargs = arg_parser.parse_known_args()
     print(args)
 
     infer_image = args.image_url
+    if not args.instance_type:
```
Comment (classicsong) [inline sagemaker/launch/launch_infer.py:178]: Same here.

File: sagemaker/launch/launch_infer.py [hunk 4]
```diff
@@ -125,61 +125,58 @@ def run_job(input_args, image, unknowargs):
               {"Key":"GraphStorm_Task", "Value":"Inference"}],
```
Comment (classicsong) [inline sagemaker/launch/launch_infer.py:125]: missing **estimator_kwargs as in launch_partition.py

File: sagemaker/launch/launch_train.py [hunk 5]
```diff
@@ -107,57 +112,49 @@ def run_job(input_args, image, unknowargs):
...
 
 if __name__ == "__main__":
-    arg_parser = parse_args()
+    arg_parser = get_train_parser()
     args, unknownargs = arg_parser.parse_known_args()
     print(args)
 
     train_image = args.image_url
 
+    if not args.instance_type:
```
Comment (classicsong) [inline sagemaker/launch/launch_train.py:157]: Same here.

File: sagemaker/launch/launch_gconstruct.py (hunk 1 above)
Comment (thvasilo) [reply sagemaker/launch/launch_gconstruct.py:129]: Right, I'll remove the `required` from the common parser to allow different default instances for each task.

**Comment by thvasilo** [comment]
@classicsong addressed all PR comments. Let me know if there's anything else you'd like to see.

**Review by classicsong — APPROVED** [review]
LGTM

**Pull request merged**